From 6a618a62a1d47e6b849e05b9a287241dfab01c01 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:03:14 -0400
Subject: [fs] btrfs: return ptr error from compression workspace

Message-id: <1407179109-13714-308-git-send-email-sandeen@redhat.com>
Patchwork-id: 87268
O-Subject: [RHEL7.1 PATCH 307/422] btrfs: return ptr error from compression workspace
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Zach Brown <zab@redhat.com>

commit 774bcb35f0d9ccb9adfd3a391328cc25523659fc
Author: Zach Brown <zab@redhat.com>
Date:   Fri May 9 17:15:09 2014 -0400

    btrfs: return ptr error from compression workspace

    The btrfs compression wrappers translated errors from workspace
    allocation to either -ENOMEM or -1.  The compression type workspace
    allocators are already returning a ERR_PTR(-ENOMEM).  Just return that
    and get rid of the magical -1.

    This helps a future patch return errors from the compression wrappers.

    Signed-off-by: Zach Brown <zab@redhat.com>
    Reviewed-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/compression.c b/fs/btrfs/compression.c
index a48cbad..149a331 100644
--- a/fs/btrfs/compression.c
+++ b/fs/btrfs/compression.c
@@ -888,7 +888,7 @@ int btrfs_compress_pages(int type, struct address_space *mapping,
 
  workspace = find_workspace(type);
  if (IS_ERR(workspace))
-  return -1;
+  return PTR_ERR(workspace);
 
  ret = btrfs_compress_op[type-1]->compress_pages(workspace, mapping,
             start, len, pages,
@@ -924,7 +924,7 @@ static int btrfs_decompress_biovec(int type, struct page **pages_in,
 
  workspace = find_workspace(type);
  if (IS_ERR(workspace))
-  return -ENOMEM;
+  return PTR_ERR(workspace);
 
  ret = btrfs_compress_op[type-1]->decompress_biovec(workspace, pages_in,
         disk_start,
@@ -946,7 +946,7 @@ int btrfs_decompress(int type, unsigned char *data_in, struct page *dest_page,
 
  workspace = find_workspace(type);
  if (IS_ERR(workspace))
-  return -ENOMEM;
+  return PTR_ERR(workspace);
 
  ret = btrfs_compress_op[type-1]->decompress(workspace, data_in,
         dest_page, start_byte,
-- 
1.7.1